src/framework/filterControl/FilterItem.jsx

```diff
@@ -0,0 +1,158 @@
+
+import React, {
+  Component,
+  PropTypes,
+} from 'react';
+
+import {
+  Filter,
+} from '../services';
+
+import Entity from '../services/string/Entity';
+import FilterForm from './forms/FilterForm.jsx';
+
+export default class FilterItem extends Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isFormOpen: false,
+    };
+
+    this.onCancelEditing = this.onCancelEditing.bind(this);
+    this.onClickAnywhere = this.onClickAnywhere.bind(this);
+    this.onEditFilter = this.onEditFilter.bind(this);
+    this.onRemoveSelectedFilter = this.onRemoveSelectedFilter.bind(this);
+    this.onUpdateFilter = this.onUpdateFilter.bind(this);
+  }
+
+  componentDidMount() {
+    document.addEventListener('click', this.onClickAnywhere);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('click', this.onClickAnywhere);
+  }
+
+  onCancelEditing() {
+    this.setState({
+      isFormOpen: false,
+    });
+  }
+
+  onClickAnywhere(event) {
+    if (
+      this.state.isFormOpen &&
+      !this.filterFormDropdown.contains(event.target) &&
+      !this.filterItemDiv.contains(event.target)
+    ) {
+      this.setState({
+        isFormOpen: false,
+      });
+    }
+  }
+
+  onEditFilter() {
+    this.setState({
+      isFormOpen: true,
+    });
+  }
+
+  onRemoveSelectedFilter() {
+    this.props.onRemoveSelectedFilter(this.props.filter);
+  }
+
+  onUpdateFilter(oldFilter, filter) {
+    this.props.onReplaceFilter(oldFilter, filter);
+    this.setState({
+      isFormOpen: false,
+    });
+  }
+
+  renderFilterForm() {
+    const onUpdateFilter =
+      filter => this.onUpdateFilter(this.props.filter, filter);
+
+    return (
+      <FilterForm
+        comparisonValue={this.props.filter.comparisonValue}
+        filterOption={this.props.filter.filterOption}
+        onAddFilter={onUpdateFilter}
+      />
+    );
+  }
+
+  renderForm() {
+    if (!this.state.isFormOpen) {
+      return undefined;
+    }
+
+    return (
+      <div
+        className="filterFormDropdown filterFormDropdown--editing"
+        ref={(div) => { this.filterFormDropdown = div; }}
+      >
+        <div className="filterFormDropdown__form">
+          <div className="filterFormDropdown__form__header">
```

Comment:
I feel like the whole component is still not well-structured. This header part is being rendered one three different ways with almost the same content. Perhaps introducing some `FilterLabel` would help cope. I'd vote for moving scope to new ticket.